test/e2e/operator_test.go

```diff
@@ -2757,9 +2828,9 @@ func newNodePortController(name types.NamespacedName, domain string) *operatorv1
 	}
 }
 
-func newHostNetworkController(name types.NamespacedName, domain string) *operatorv1.IngressController {
+func newHostNetworkController(name types.NamespacedName, domain string, strategy *operatorv1.HostNetworkStrategy, placement *operatorv1.NodePlacement) *operatorv1.IngressController {
```

Comment:
In other tests, we have the caller mutate the object that `newFooController` returns:
```go
	ic := newPrivateController(icName, domain)
	ic.Spec.Foo = Bar
	if err := kclient.Create(context.TODO(), ic); err != nil {
```
That makes the test code more explicit and the helper more generic.  Could you do likewise for `TestHostNetworkPortBinding`?